Sources/SmartInsights/UpdateAvailableInsight.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  UpdateAvailableInsight.swift
+//  Diagnostics
+//
+//  Created by Antoine van der Lee on 09/02/2022.
+//  Copyright © 2019 WeTransfer. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+struct UpdateAvailableInsight: SmartInsightProviding {
+    
+    let name = "Update available"
+    let result: InsightResult
+    
+    private var urlSessionAppMetadataPublisher: (URL) -> AnyPublisher<AppMetadataResults, Error> = { url in
+        URLSession.shared
+            .dataTaskPublisher(for: url)
+            .map(\.data)
+            .decode(type: AppMetadataResults.self, decoder: JSONDecoder())
+            .eraseToAnyPublisher()
+    }
+    
+    init?(bundleIdentifier: String? = Bundle.main.bundleIdentifier, currentVersion: String = Bundle.appVersion, appMetadataPublisher: AnyPublisher<AppMetadataResults, Error>? = nil) {
+        guard let bundleIdentifier = bundleIdentifier else { return nil }
+        let url = URL(string: "https://itunes.apple.com/br/lookup?bundleId=\(bundleIdentifier)")!
+        
+        let group = DispatchGroup()
+        group.enter()
+        
+        var appMetadata: AppMetadata?
+        let publisher = appMetadataPublisher ?? urlSessionAppMetadataPublisher(url)
+        let cancellable = publisher
+            .sink { _ in
+                group.leave()
+            } receiveValue: { result in
+                appMetadata = result.results.first
+            }
+        
+        /// Set a timeout of 1 second to prevent the call from taking too long unexpectedly.
+        /// Though: the request should be super fast since it's a small resource.
+        let result = group.wait(timeout: .now() + .seconds(1))
```

Comment:
Note: I'm very aware this is not the most beautiful solution here. Though: the other solution would be to make creating a report asynchronously which will require a lot more code changes. I decided to keep it simple here, since it will at most take 1 second to generate. This way we don't require to show spinners and stuff while generating a report.

Reply:
We could also do it async in the JS and the result will be based on the time that somebody is looking at the report.

Reply:
I thought about that, but it's a bit harder to implement. Also: it's about the status at the time of reporting. If at that point no new update was available, the user was also not able to fix it by updating. I guess there's still value in having it through js since we might have fixed an issue already, but it's not straightforward to implement right now